core/types/block.go

```diff
@@ -249,6 +129,10 @@ func NewBlock(
 // CopyHeader creates a deep copy of a block header.
 func CopyHeader(h *Header) *Header {
 	cpy := *h
+	extras.Header.Set(&cpy, &HeaderExtra{})
+	cpyExtra := HeaderExtras(&cpy)
+	*cpyExtra = *HeaderExtras(h)
```

Comment:
nit a bit less convoluted I think:

```suggestion
	extra := HeaderExtras(h)
	cpyExtra := &HeaderExtra{
		ExtDataHash: extra.ExtDataHash,
	}
	cpy = *WithHeaderExtras(&cpy, cpyExtra)
```

You could also define `hExtra` instead of `extra` here to save one line below